### ISSUE TITLE
BUG: If netCDF file containing regional average data does not yet exist, aospy crashes when trying to save

### ISSUE THREAD

**Issue by spencerkclark** [spencerahill/aospy#99]
@spencerahill I think in the except statement in [line 818 of calc.py](https://github.com/spencerahill/aospy/blob/develop/aospy/calc.py#L818) we simply need to add `IOError`, so that when this exception is raised a fresh Dataset can be created (this code already exists it just needs to be called). I can put together a short PR if this makes sense.

Here is the traceback for reference:

```
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
/home/Spencer.Clark/aospy_example/main.py in <module>()
     23                                      **parameters)
     24             calc = Calc(calc_int)
---> 25             calc.compute()

/home/Spencer.Clark/aospy/aospy/calc.pyc in compute(self, save_to_scratch, save_to_archive)
    806         for dtype_time, data in reduced.items():
    807             self.save(data, dtype_time, dtype_out_vert=self.dtype_out_vert,
--> 808                       scratch=save_to_scratch, archive=save_to_archive)
    809
    810     def _save_to_scratch(self, data, dtype_out_time):

/home/Spencer.Clark/aospy/aospy/calc.pyc in save(self, data, dtype_out_time, dtype_out_vert, scratch, archive)
    870         self._update_data_out(data, dtype_out_time)
    871         if scratch:
--> 872             self._save_to_scratch(data, dtype_out_time)
    873         if archive:
    874             self._save_to_archive(dtype_out_time)

/home/Spencer.Clark/aospy/aospy/calc.pyc in _save_to_scratch(self, data, dtype_out_time)
    815         if 'reg' in dtype_out_time:
    816             try:
--> 817                 reg_data = xr.open_dataset(path)
    818             except (EOFError, RuntimeError):
    819                 reg_data = xr.Dataset()

/nbhome/skc/anaconda2/lib/python2.7/site-packages/xarray/backends/api.pyc in open_dataset(filename_or_obj, group, decode_cf, mask_and_scale, decode_times, concat_characters, decode_coords, engine, chunks, lock, drop_variables)
    208                                              allow_remote=True)
    209             if engine == 'netcdf4':
--> 210                 store = backends.NetCDF4DataStore(filename_or_obj, group=group)
    211             elif engine == 'scipy':
    212                 store = backends.ScipyDataStore(filename_or_obj)

/nbhome/skc/anaconda2/lib/python2.7/site-packages/xarray/backends/netCDF4_.pyc in __init__(self, filename, mode, format, group, writer, clobber, diskless, persist)
    186         ds = nc4.Dataset(filename, mode=mode, clobber=clobber,
    187                          diskless=diskless, persist=persist,
--> 188                          format=format)
    189         with close_on_error(ds):
    190             self.ds = _nc4_group(ds, group, mode)

netCDF4/_netCDF4.pyx in netCDF4._netCDF4.Dataset.__init__ (netCDF4/_netCDF4.c:12624)()

IOError: No such file or directory
```

**Comment by spencerahill** [comment]
Good catch.  Yes, that would be great.

Is it feasible to write tests on Calc functionality like this given the current state of Calc?  Or should we just call this a quick fix and deal with the testing once we refactor?  Ideally we add tests for each new bug we find, but since this one is so small and Calc is so hairy, I don't want it to be a nuisance.

**Comment by spencerkclark** [comment]
I think we would need to create a Calc object to test on since `_save_to_scratch` is an instance method and depends on some internal Calc attributes.  We should probably think carefully about how / if we should create Calc objects to test certain methods (otherwise we'll end up with messy test cases), so if it's OK maybe we can put this off until after the refactor?

Maybe in the future we could consider making this a static method so that it could be more easily tested?

**Comment by spencerahill** [comment]
> maybe we can put this off until after the refactor?

Agreed.  I'd say that for any one-off fixes in Calc such as this between now and the refactor, let's not bother with mocking up a Calc object, etc. to test.

> Maybe in the future we could consider making this a static method so that it could be more easily tested?

C.f. #33, I think all of the write-to-disk functionality should be bundled together in some way separate from `Calc`, either a module or module + class(es)...the initial though in #33 is a `DataExporter` class.

**Comment by spencerahill** [comment]
Closed by #100
